train.py

```diff
@@ -159,9 +159,15 @@ def loss_fn(pred, labels):
         models_parallelize_fns[model_name](model, world_mesh, parallel_dims, job_config)
 
         # move sharded model to CPU/GPU and initialize weights via DTensor
-        init_device = "cpu" if job_config.checkpoint.create_seed_checkpoint else "cuda"
+        init_device = (
+            "cpu"
+            if job_config.checkpoint.create_seed_checkpoint
+            or job_config.training.enable_cpu_offload
```

Comment:
Don't we need to do the same for the PP case (several lines above)? Or are we assuming if PP is used, CPU offloading is not an option?

Reply:
I test it with pp.
Based on `llama2_7b.toml`, cpu_offload also works but has a much obvious latency in training.
<img width="587" alt="Screenshot 2024-10-25 at 3 40 25 PM" src="https://github.com/user-attachments/assets/5faacd43-7964-471c-ac0c-b1cf2903b0a8">